### PR TITLE
test: Bring up test coverage for SentryHttpTransport

### PR DESCRIPTION
**Pull request by philipphofmann**
Add more tests for SentryHttpTransport so that the coverage is almost 100%. Extracted duplicated code for sending a request from `sendEvent` and `sendEnvelope` to a new method.

### CODE REVIEW

File: Sources/Sentry/SentryHttpTransport.m [hunk 1]
```diff
@@ -52,7 +52,7 @@ - (void)    sendEvent:(SentryEvent *)event
     
     NSError *requestError = nil;
     // TODO: We do multiple serializations here, we can improve this
-    SentryNSURLRequest *request = [[SentryNSURLRequest alloc] initStoreRequestWithDsn:self.options.dsn
+    NSURLRequest *request = [[SentryNSURLRequest alloc] initStoreRequestWithDsn:self.options.dsn
```
Comment (HazAT) [inline Sources/Sentry/SentryHttpTransport.m:55]: Why this change?

Comment (philipphofmann) [reply Sources/Sentry/SentryHttpTransport.m:55]: Because of the [principle of least knowledge](https://en.wikipedia.org/wiki/Law_of_Demeter). `sendRequest` doesn't need to know it is a `SentryNSURLRequest`. Only knowing about `NSURLRequest` it is more flexible.
